sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/BufferedHttpResponse.java

```diff
@@ -28,12 +36,13 @@
     public BufferedHttpResponse(HttpResponse innerHttpResponse) {
         super(innerHttpResponse.getRequest());
         this.innerHttpResponse = innerHttpResponse;
-        this.cachedBody = FluxUtil.collectBytesFromNetworkResponse(innerHttpResponse.getBody(),
-            innerHttpResponse.getHeaders())
-            .map(ByteBuffer::wrap)
-            .flux()
-            .cache()
-            .map(ByteBuffer::duplicate);
+        this.cachedBody = innerHttpResponse.getBody()
+            .map(buffer -> {
+                cachedBodySize.addAndGet(buffer.remaining());
+                return ByteBuffer.wrap(FluxUtil.byteBufferToArray(buffer));
```

Comment:
I'm curious if we have done any investigation into this approach of copying the `ByteBuffer` into a `byte[]`, and then wrapping this `byte[]` as a new `ByteBuffer`, and if there are better approaches? For example, there are various approaches detailed [here](https://stackoverflow.com/questions/3366925/deep-copy-duplicate-of-javas-bytebuffer).